examples/bigcommerce-catalyst/components/product-carousel/product-carousel.makeswift.ts

```diff
@@ -0,0 +1,44 @@
+import dynamic from 'next/dynamic';
+
+import { Combobox, TextInput } from '@makeswift/runtime/controls';
+import { forwardNextDynamicRef } from '@makeswift/runtime/next';
+
+import { runtime } from '~/lib/makeswift/runtime';
+
+type Category = {
+  entityId: number;
+  name: string;
+  path: string;
+}
+
+export const props = {
+  title: TextInput({ label: 'Title', defaultValue: 'Products', selectAll: true }),
+  products: Combobox({
+    label: "Category",
+    async getOptions(query) {
+      const { categories }: { categories: Category[] } = await fetch('/api/categories').then((res) => res.json())
+
+      const mappings = await Promise.all(categories
+        .map(async (category) => {
+          const { products } = await fetch(`/api/quick-search?categoryId=${category.entityId}&imageHeight=500&imageWidth=500&limit=10`).then((res) => res.json())
+
+          return {
+            id: category.entityId.toString(),
+            label: category.name,
+            value: products,
```

Comment:
Instead of storing the products in the data like this I thought we were going to pass the category ID to the component and then fetch products from the component dynamically. Otherwise these products will be set when the prop is set and won't change it the catalog data changes. And as for making it work well with SSR we'd do an SSR pre-pass.